internal/controllers/lvmcluster/status.go

```diff
@@ -0,0 +1,161 @@
+package lvmcluster
+
+import (
+	"fmt"
+
+	lvmv1alpha1 "github.com/openshift/lvm-operator/api/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	ReasonReconciliationInProgress  = "ReconciliationInProgress"
+	MessageReconciliationInProgress = "Reconciliation is still in progress"
+
+	ReasonResourcesSynced  = "ResourcesSynced"
+	MessageResourcesSynced = "Reconciliation is successfully complete"
+
+	ReasonResourcesSyncFailed        = "ResourcesSyncFailed"
+	MessageReasonResourcesSyncFailed = "Reconciliation is failed with %v"
+
+	ReasonVGReadinessInProgress  = "VGReadinessInProgress"
+	MessageVGReadinessInProgress = "VG readiness check is still in progress"
+
+	ReasonVGsFailed  = "VGsFailed"
+	MessageVGsFailed = "One or more vgs are failed"
+
+	ReasonVGsDegraded  = "VGsDegraded"
+	MessageVGsDegraded = "One or more vgs are degraded"
+
+	ReasonVGsReady  = "VGsReady"
+	MessageVGsReady = "All the VGs are ready"
+)
+
+func computeInitialConditions() []lvmv1alpha1.Condition {
+	var conditions []lvmv1alpha1.Condition
+	reconciliationCondition := computeResourcesSyncedCondition(true, nil)
+	conditions = append(conditions, reconciliationCondition)
+	vgReadinessCondition := computeVGReadinessCondition(true, nil)
+	conditions = append(conditions, vgReadinessCondition)
+
+	return conditions
+}
+
+func computeResourcesSyncedCondition(init bool, reconciliationError error) lvmv1alpha1.Condition {
+	condition := lvmv1alpha1.Condition{
+		Type: lvmv1alpha1.ResourcesSynced,
+	}
+	if init {
```

Comment:
This is not needed if you slightly restructure the code. You can initialize them by default and then modify the existing condition in later parts of the code. There is no need to compute them from scratch everytime.